src/translator/helper.ts

```diff
@@ -52,3 +64,66 @@ export const createConditionalRenderingObject = (
 
     return Object.keys(conditionObj).length === 0 ? [] : [conditionObj];
 };
+
+export const translateConditionalRenderingObject = (
+    conditions: ISchemaCondition[],
+    internalId: string
+) => {
+    return conditions.reduce((translatedConditions, condition) => {
+        Object.entries(condition).forEach(([key, value]) => {
+            value
+                .filter((obj) => !("filled" in obj))
+                .forEach((condition) => {
+                    const [comparator, compValue] =
+                        Object.entries(condition)[0];
+                    translatedConditions.push({
+                        fieldKey: key,
+                        comparator: ELEMENT_CONDITION_TYPES[comparator],
+                        value: compValue,
+                        internalId,
+                    });
+                });
+        });
+        return translatedConditions;
+    }, []);
+};
+
+export const translateSchemaBasedOnType = (
```

Comment:
since we are using `parseSchema` in the imperative handle, let's keep the naming consistent, replace `translateSchema` prefixes with `parseSchema`

Reply:
can we update the rest of the names?
- updateTranslatedElements
- translateToElement
- translateConditionalRenderingObject
- translatedConditions
- schemaToTranslate
- translatedElements